Provider/src/FirebirdSql.EntityFrameworkCore.Firebird/Migrations/FbMigrationsSqlGenerator.cs

```diff
@@ -15,15 +15,161 @@
+			}
+		}
+
+		protected override void Generate(CreateTableOperation operation, IModel model, MigrationCommandListBuilder builder, bool terminate)
+		{
+			base.Generate(operation, model, builder, false);
+			if (terminate)
+			{
+				builder.Append(Dependencies.SqlGenerationHelper.StatementTerminator);
+				EndStatement(builder);
+
+				var columns = operation.Columns.Where(p => !p.IsNullable && string.IsNullOrWhiteSpace(p.DefaultValueSql) && p.DefaultValue == null);
+				foreach (var column in columns)
+				{
+					var colAnnotation = column.FindAnnotation(FbAnnotationNames.ValueGenerationStrategy);
+					if (colAnnotation != null)
+					{
+						var typeSequence = IsSequenceIdentityOrTrigger(colAnnotation.Value as FbValueGenerationStrategy?);
+						if (typeSequence == FbValueGenerationStrategy.SequenceTrigger)
+						{
+							GenerateSequenceOnTrigger(builder, column.Table, column.Name);
+						}
+					}
+				}
+			}
+		}
+
+		protected override void ColumnDefinition(string schema, string table, string name, Type clrType, string type, bool? unicode, int? maxLength, bool rowVersion, bool nullable, object defaultValue, string defaultValueSql, string computedColumnSql, IAnnotatable annotatable, IModel model, MigrationCommandListBuilder builder)
+		{
+			var valueGenerationStrategy = annotatable[FbAnnotationNames.ValueGenerationStrategy] as FbValueGenerationStrategy?;
+			var typeSequence = IsSequenceIdentityOrTrigger(valueGenerationStrategy);
+
+			builder.Append(Dependencies.SqlGenerationHelper.DelimitIdentifier(name))
+				   .Append(" ")
+				   .Append(type ?? GetColumnType(schema, table, name, clrType, unicode, maxLength, rowVersion, model));
+
+			if (!nullable)
+			{
+				switch (typeSequence)
+				{
+					case FbValueGenerationStrategy.IdentityColumn:
+						builder.Append(" GENERATED BY DEFAULT AS IDENTITY NOT NULL");
+						break;
+					case FbValueGenerationStrategy.None:
+						builder.Append(" NOT NULL");
+						break;
+				}
+			}
+			else
+			{
+				if (defaultValueSql != null)
+				{
+					builder.Append(" DEFAULT ")
+						   .Append(defaultValueSql);
+				}
+				else if (defaultValue != null)
+				{
+					var defaultValueLiteral = Dependencies.TypeMapper.GetMapping(clrType);
+					builder.Append(" DEFAULT ")
+						   .Append(defaultValueLiteral.GenerateSqlLiteral(defaultValue));
+				}
+
+				if (!nullable)
+					builder.Append(" NOT NULL");
+			}
+		}
+
+		protected override void ForeignKeyAction(ReferentialAction referentialAction, MigrationCommandListBuilder builder)
+		{
+			if (referentialAction == ReferentialAction.Restrict)
+				builder.Append("NO ACTION");
+			else
+				base.ForeignKeyAction(referentialAction, builder);
+		}
+
+		private FbValueGenerationStrategy IsSequenceIdentityOrTrigger(FbValueGenerationStrategy? valueGenerate)
+		{
+			return valueGenerate ?? FbValueGenerationStrategy.None;
+		}
+
+		private void GenerateSequenceOnTrigger(MigrationCommandListBuilder builder, string table, string column)
```

Comment:
I think this should use interface similar what's [here](https://github.com/cincuranet/FirebirdSql.Data.FirebirdClient/blob/master/Provider/src/EntityFramework.Firebird/DefaultFbMigrationSqlGeneratorBehavior.cs). This is often opinionated and should be easily replaceable.